src/lib/swapper/utils.ts

```diff
@@ -2,26 +2,33 @@ import type { SwapErrorRight } from '@shapeshiftoss/swapper'
 import { makeSwapErrorRight, SwapErrorType, SwapperName } from '@shapeshiftoss/swapper'
 import type { Result } from '@sniptt/monads'
 import { Err, Ok } from '@sniptt/monads'
-import type { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios'
-import type { ISetupCache } from 'axios-cache-adapter'
-import { setupCache } from 'axios-cache-adapter'
+import Axios, { type AxiosInstance, type AxiosRequestConfig, type AxiosResponse } from 'axios'
+import { setupCache } from 'axios-cache-interceptor'
 import type { FeatureFlags } from 'state/slices/preferencesSlice/preferencesSlice'
 
 import { isCrossAccountTradeSupported } from '../../state/helpers'
 import { AsyncResultOf, isTruthy } from '../utils'
 
-const getRequestFilter = (cachedUrls: string[]) => (request: Request) =>
-  !cachedUrls.some(url => request.url.includes(url))
+const getRequestFilter = (cachedUrls: string[]) => (request: AxiosRequestConfig) =>
+  !cachedUrls.some(url => request.url?.includes(url))
 
-export const createCache = (maxAge: number, cachedUrls: string[]): ISetupCache => {
+export const createCache = (
+  maxAge: number,
+  cachedUrls: string[],
+  axiosConfig: AxiosRequestConfig,
+): AxiosInstance => {
   const filter = getRequestFilter(cachedUrls)
-  return setupCache({
-    maxAge,
-    exclude: { filter, query: false },
-    clearOnStale: true,
-    readOnError: false,
-    readHeaders: false,
+  const axiosInstance = Axios.create(axiosConfig)
+
+  setupCache(axiosInstance, {
+    ttl: maxAge,
+    cachePredicate: cacheResponse => filter(cacheResponse.config),
+    interpretHeader: false,
+    staleIfError: true,
+    cacheTakeover: false,
```

Comment:
[cacheTakeover](https://axios-cache-interceptor.js.org/config/request-specifics#cache-cachetakeover): 
```
When your requested routes includes Cache-Control in their responses, you may end up with we and your browser caching the response, resulting in a double layer of cache.

This option solves this by including some predefined headers in the request, that should tell any client / adapter to not cache the response, thus only we will cache it.
```

That's theoretically very nice-to-have to avoid double caching, however, this assumes *all* endpoints handle `Cache-Control`, `Pragma`, and `Expires` headers gracefully:

> When you encounter this error, you need to make sure Cache-Control, Pragma and Expires headers are included into your server’s Access-Control-Allow-Headers CORS configuration

Seems like most endpoints are handling these, but not THOR.

Having this option to the default `true` will result in CORS errors and failed requests, because the THOR endpoint doesn't gracefully handle those options:

```json
{
  'Cache-Control': 'no-cache',
  'Pragma': 'no-cache',
  'Expires': '0',
}
  ```
  
and won't respond with the `Access-Control-Allow-Origin` header.